internal/report/prometheus.go

```diff
@@ -57,36 +74,42 @@ var (
 		Help:      "Sanitizer score results for resource groups.",
 	},
 		[]string{
+			"cluster",
+			"namespace",
 			"title",
 		})
-	errs = prometheus.NewGauge(prometheus.GaugeOpts{
+	errs = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: namespace,
 		Name:      "errors",
 		Help:      "Errors while sanitizing the cluster.",
-	})
+	},
+		[]string{
+			"cluster",
+			"namespace",
+		})
 )
 
-func prometheusMarshal(b *Builder, address *string) *push.Pusher {
+func prometheusMarshal(b *Builder, address *string, cluster, namespace string) *push.Pusher {
 	pusher := newPusher(address)
 
-	score.Set(float64(b.Report.Score))
-	grade.Set(float64(gradeToNumber(b.Report.Grade)))
-	errs.Set(float64(len(b.Report.Errors)))
+	score.WithLabelValues(cluster, namespace).Set(float64(b.Report.Score))
+	grade.WithLabelValues(cluster, namespace, b.Report.Grade).Set(float64(gradeToNumber(b.Report.Grade)))
+	errs.WithLabelValues(cluster, namespace).Set(float64(len(b.Report.Errors)))
 
 	for _, section := range b.Report.Sections {
 		for i, v := range section.Tally.counts {
 			switch i {
 			case 0:
-				sanitizersOk.WithLabelValues(section.Title).Set(float64(v))
```

Comment:
@eminugurkenar Leveraging tags vs metrics for every single sanitizer level will also simplify this code and allow us to introduce new levels in the future without having to introduce new metric types.

Reply:
now they are one single metric with "level" label. Metric name "popeye_sanitizers_findings_total" now. What do you think about it ?